### PR TITLE
feat(tooltip): add tippyjs wrapper and remove decorator

### LINKED ISSUE #3316: Help: wrong alignment for the bottom/top left alignment
### Current behaviour

When the tooltip is set to be shown `top`/`bottom` + `left` alignment - that the tooltip point arrow is shown not in the middle of the icon 

Screenshot 2020-10-21 at 13 46 53
Screenshot 2020-10-21 at 13 56 34

### Expected behaviour

The tooltip pointer should be aligned in the middle of the icon.

Screenshot 2020-10-21 at 14 00 19

### Reproducible example

[An example](https://carbon.sage.com/iframe.html?id=help--default&knob-tooltipPosition=top&knob-tooltipAlign=left)

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 44.1.0 |

### LINKED ISSUE #3514: NumeralDate - Tooltip flash when hover over icon
### Current behaviour

Tooltip flashes when hovered over it when the icon is inside of a input

### Expected behaviour

Either there should be no tooltip and it should be displayed when the input is selected or the tooltip should show on hover and not flash

### Reproducible example

https://carbon.sage.com/?path=/story/design-system-numeral-date--date-component-has-warning
hover over the icon inside the input

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 54.3
| carbon-factory   |
| react-scripts    | 
| React            |
| Browser          | All
| npm              | 
| Operating System | Mac OS Catalina

### LINKED ISSUE #3575: Tooltip message in scrollable form
### Current behaviour
When scrolling in a form where there is a tooltip, the message of the tooltip does not follow the icon.

### Expected behaviour
When scrolling in a form where there is a tooltip, the message of the tooltip have to follow the icon.

### Reproducible example
See capture below :
image

### Your environment
| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 54.6.0
| React            | 16.13.1
| Browser          | Chrome
| npm              | 6.14.2
| Operating System | Windows 10

### PR DESCRIPTION
**Pull request by edleeks87**
fixes #3575
fixes #3514
fixes #3316 

### Proposed behaviour

The `TooltipDecorator` has been deleted and the functionality it provided is now achieved by adding
`tippyjs`, the `align` props are no longer available as tippy provides that dynamically. Some style
changes have been made to bring the component into alignment with what is described in the DLS.

BREAKING CHANGE: The `TooltipDecorator` has been removed, and the interface for `Tooltip` has been
updated

The Tooltip now remains fixed to the target component on scroll, positions dynamically if there is no space in the window and pointer dynamically updates it's alignment to remain pointed at target.

Fixed a bug in `TextEditor` where if the style controls were focused and then the user clicks on the editor again focus was not being properly transferred away from the control.

Refactored components to be functional now that the decorator has been removed, this was needed as `tippyjs` needs target components to support `forwardRef`

### Current behaviour

The `Tooltip` relies on the `TooltipDecorator` to implement the display, alignment and positioning for example. Any component needing to use it has to be class based.
All the issues raised: `Tooltip`s do not remain fixed to the target component when a user scrolls, some of the alignment and positioning combinations do not work, the `Tooltip` will render offscreen if there is no space in the window.

Bug in `TextEditor`

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context

image

image

image

image

image

### Testing instructions

https://codesandbox.io/s/carbon-quickstart-forked-147w2
In popover container: https://codesandbox.io/s/carbon-quickstart-forked-x82pr?file=/src/index.js

Updates can also be tested in any story that has a tooltip or a validation/ help icon (`text-editor`, `validation-icon`, `help`, all inputs with validation, `tooltip`)

### CODE REVIEW

**Review by KatarzynaQA — APPROVED** [review]
Cypress tests approved

File: src/components/icon/icon.component.js [hunk 1]
```diff
@@ -1,65 +1,70 @@
...
+      bgTheme,
+      className,
+      color,
+      disabled,
+      fontSize,
+      iconColor,
+      type,
+      ml,
+      mr,
+      ...props
```
Comment (sjohnson-sage) [inline src/components/icon/icon.component.js:21]: This should really be `...rest` rather than `...props`

File: src/components/icon/icon.component.js [hunk 2]
```diff
@@ -1,65 +1,70 @@
...
-        aria-label={this.props.ariaLabel}
-      />,
-      this.tooltipHTML,
-    ];
+        className={className || null}
+        {...tagComponent("icon", props)}
+        data-element={iconType()}
+        mr={mr}
+        ml={ml}
+        {...props}
```
Comment (sjohnson-sage) [inline src/components/icon/icon.component.js:63]: Do we really want to pass all the rest of the props down to `StyledIcon` here? Anything that `StyledIcon` needs should be passed down explicitly and nothing else should be passed to it - it'll help avoid problems like we're having with `FormField` etc at the moment

File: src/components/portal/__spec__.js [hunk 3]
```diff
@@ -17,12 +17,7 @@ describe("Portal", () => {
     beforeEach(() => {
       wrapper = mount(
         <Portal>
-          <Icon
-            tooltipMessage="Test"
-            tooltipAlign="left"
-            tooltipPosition="top"
-            type="tick"
-          />
+          <Icon tooltipMessage="Test" tooltipPosition="top" type="tick" />
```
Comment (sjohnson-sage) [inline src/components/portal/__spec__.js:20]: I think we need to document the `tooltipMessage` and `tooltipPosition` props on `Icon`

Comment (sjohnson-sage) [reply src/components/portal/__spec__.js:20]: Actually it won't have these props will it, shouldn't this be `Icon` wrapped in `Tooltip`?

File: src/components/text-editor/__internal__/text-editor.style.js [hunk 4]
```diff
@@ -54,6 +54,8 @@ StyledEditorContainer.defaultProps = {
 const StyledEditorOutline = styled.div`
   ${({ theme, isFocused, hasError }) => css`
     outline: none;
+    min-width: 320px;
```
Comment (sjohnson-sage) [inline src/components/text-editor/__internal__/text-editor.style.js:57]: Why is this needed for the tooltips?

File: src/components/portal/__spec__.js (hunk 3 above)
Comment (edleeks87) [reply src/components/portal/__spec__.js:20]: Yeah agreed I've refactored it to render Tooltip within Icon now

File: src/components/text-editor/__internal__/text-editor.style.js (hunk 4 above)
Comment (edleeks87) [reply src/components/text-editor/__internal__/text-editor.style.js:57]: It isn't I noticed a style issue in `TextEditor` when the window resized so added a fix here as it hadn't been raised and was a one liner

Comment (edleeks87) [reply src/components/text-editor/__internal__/text-editor.style.js:57]: In retrospect might have been worth doing it separately as would have reduced the files by one at least 😆

File: src/components/icon/icon.component.js (hunk 2 above)
Comment (edleeks87) [reply src/components/icon/icon.component.js:67]: Yeah good point I'll sort it once the reviews have finished

File: src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.component.js [hunk 5]
```diff
@@ -1,65 +1,54 @@
...
+  (
+    {
+      onKeyDown,
+      onMouseDown,
+      activated,
+      ariaLabel,
+      tabbable,
+      children,
+      tooltipMessage,
+      ...props
```
Comment (sjohnson-sage) [inline src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.component.js:15]: This one should be `...rest` as well

File: src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.d.ts [hunk 6]
```diff
@@ -6,9 +6,6 @@ export interface ToolbarButtonProps {
   activated: boolean;
   onKeyDown: (event: React.KeyboardEvent<HTMLElement>) => void;
   onMouseDown: (event: React.MouseEvent<HTMLElement>) => void;
-  onMouseOver: (event: React.MouseEvent<HTMLElement>) => void;
-  onMouseLeave: (event: React.MouseEvent<HTMLElement>) => void;
-  tooltipVisible?: boolean;
```
Comment (sjohnson-sage) [inline src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.d.ts:11]: I think we want to keep `tooltipVisible` in here, and add `tooltipMessage` as well

File: src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.spec.js [hunk 7]
```diff
@@ -80,36 +81,4 @@ describe("ToolbarButton", () => {
...
-        tooltipVisible: false,
-      };
-      return (
-        <ToolbarButton {...defaultProps} {...props}>
-          foo
-        </ToolbarButton>
-      );
-    };
-
-    it("is displayed when `tooltipVisible` is true", () => {
```
Comment (sjohnson-sage) [inline src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.spec.js:103]: Are these not still valid tests? `tooltipVisible` is still a prop of `ToolbarButton` from looking above

File: src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.d.ts (hunk 6 above)
Comment (edleeks87) [reply src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.d.ts:11]: Yeah good point I forgot to add them back in when I refactored it earlier

File: src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.spec.js (hunk 7 above)
Comment (edleeks87) [reply src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.spec.js:103]: Yeah same as above forgot to add back in, thanks will do when the reviews are complete

File: src/components/text-editor/__internal__/toolbar/toolbar.component.js [hunk 8]
```diff
@@ -59,89 +58,132 @@ const Toolbar = ({
         setTabbable(true);
       } else if (Events.isLeftKey(ev)) {
         if (focusIndex === null || focusIndex === 0) {
+          controlRefs[3].current.focus();
```
Comment (sjohnson-sage) [inline src/components/text-editor/__internal__/toolbar/toolbar.component.js:61]: Is this all unrelated to tooltips as well?

File: src/components/tile-select/tile-select.stories.mdx [hunk 9]
```diff
@@ -91,7 +92,7 @@ These props are then internally distributed on each of the `TileSelects` making
             disabled
             title="Title" 
             subtitle="Subtitle"
-            titleAdornment={<Icon type='info' tooltipMessage="Short and non descriptive message" />}
+            titleAdornment={<Tooltip message="Short and non descriptive message"><Icon type="info" /></Tooltip>}
```
Comment (sjohnson-sage) [inline src/components/tile-select/tile-select.stories.mdx:95]: This can change back after your refactor I think

Comment (sjohnson-sage) [reply src/components/tile-select/tile-select.stories.mdx:95]: Same with the others in this file

File: src/components/tooltip/index.d.ts [hunk 10]
```diff
@@ -0,0 +1,2 @@
+export { default } from './tooltip';
```
Comment (sjohnson-sage) [inline src/components/tooltip/index.d.ts:1]: This should be `export { default } from './tooltip.component';` I think

File: src/components/tooltip/index.d.ts [hunk 11]
```diff
@@ -0,0 +1,2 @@
...
+export * from './tooltip';
```
Comment (sjohnson-sage) [inline src/components/tooltip/index.d.ts:2]: Why are we exporting everything here?

File: src/components/tooltip/tooltip.component.js [hunk 12]
```diff
@@ -1,70 +1,97 @@
...
+      children,
+      isVisible,
+      position = "top",
+      message,
+      type,
+      size = "M",
+      isPartOfInput,
+      inputSize,
+      id,
+      ...props
```
Comment (sjohnson-sage) [inline src/components/tooltip/tooltip.component.js:20]: `...rest`

File: src/components/text-editor/__internal__/toolbar/toolbar.component.js (hunk 8 above)
Comment (edleeks87) [reply src/components/text-editor/__internal__/toolbar/toolbar.component.js:61]: Yeah it was causing a bug in the tooltip behaviour so needed to be done either in this PR or be merged before it at least

File: src/components/tile-select/tile-select.stories.mdx (hunk 9 above)
Comment (edleeks87) [reply src/components/tile-select/tile-select.stories.mdx:95]: Ah yep good spot thanks

File: src/components/tooltip/tooltip.component.js [hunk 13]
```diff
@@ -1,70 +1,97 @@
...
 
-  get tooltipHTML() {
-    const { children, onMouseEnter, onMouseLeave, ...commonProps } = this.props;
+      const pointer = (
+        <StyledPointer
+          key="pointer"
+          type={type}
+          {...attrs}
+          position={currentPosition}
+          data-popper-arrow=""
```
Comment (sjohnson-sage) [inline src/components/tooltip/tooltip.component.js:36]: Is this meant to be an empty string?

Comment (edleeks87) [reply src/components/tooltip/tooltip.component.js:36]: Yeah that's how it is in the docs https://github.com/atomiks/tippyjs-react#headless-tippy

File: src/components/tooltip/tooltip.component.js [hunk 14]
```diff
@@ -1,70 +1,97 @@
...
+  /** Whether to to show the Tooltip */
+  isVisible: PropTypes.bool,
+  /** Sets position of the tooltip */
+  position: PropTypes.oneOf(["top", "bottom", "left", "right"]),
+  /** Defines the message type */
+  type: PropTypes.string,
+  /** Children elements */
+  children: PropTypes.node.isRequired,
+  /** Defines the size of the tooltip content */
+  size: PropTypes.oneOf(["M", "L"]),
```
Comment (sjohnson-sage) [inline src/components/tooltip/tooltip.component.js:90]: Maybe this should be `default` and `L` (or `large`) - `M` doesn't really mean anything when there's no small

File: src/components/validations/validation-icon.component.js [hunk 15]
```diff
@@ -48,22 +50,31 @@ const ValidationIcon = ({
...
         type={validationType}
-        size={size}
-        tabIndex={tabIndex}
-        tooltipPosition={tooltipPosition}
-        tooltipAlign="center"
-        isThemeModern
+        message={validationMessage}
+        isVisible={
+          hasFocus ||
+          hasMouseOver ||
```
Comment (sjohnson-sage) [inline src/components/validations/validation-icon.component.js:63]: `hasMouseOver` and `hasHover` are the same thing I think

Comment (sjohnson-sage) [reply src/components/validations/validation-icon.component.js:63]: Oh sorry, `hasMouseOver` is set to `groupHasMouseOver` so I think you can remove one of them

File: src/components/validations/validation-icon.component.js [hunk 16]
```diff
@@ -32,6 +33,7 @@ const ValidationIcon = ({
     hasFocus: groupHasFocus,
     hasMouseOver: groupHasMouseOver,
   } = useContext(InputGroupContext);
+  const [hasHover, setHasHover] = React.useState(false);
```
Comment (sjohnson-sage) [inline src/components/validations/validation-icon.component.js:36]: Why do we need `hasHover` when we have `hasMouseOver` from the `InputContext`?

**Review by sjohnson-sage — COMMENTED** [review]
Good work mate, not an easy task! Just a few comments from me to sort really

File: src/components/tooltip/tooltip.component.js (hunk 14 above)
Comment (edleeks87) [reply src/components/tooltip/tooltip.component.js:90]: Yeah could do, I chose the naming convention described in the DLS and I know we've used the short form for sizes in other components recently. I'll check with UX as if there's not going to be other size variants then it makes sense to simplify the interface the way you suggest, or at least maybe go with `medium` and `large`

File: src/components/validations/validation-icon.component.js (hunk 16 above)
Comment (edleeks87) [reply src/components/validations/validation-icon.component.js:36]: `hasMouseOver` is from the context for the input, the tooltip needs to also trigger when the `ValidationIcon` has a `mouseOver` or `focus` event. I think the naming it `hasHover, setHasHover` is really poor on my part as it doesn't really reflect what it's doing, I'll rename it to something like `internalTooltipTrigger` and `setInternalTooltipTrigger`

File: src/components/icon/icon.stories.js [hunk 17]
```diff
@@ -19,16 +19,8 @@ export default {
 };
 
 const commonKnobs = () => {
-  const tooltipMessage = text("tooltipMessage", "");
   return {
-    tooltipMessage,
```
Comment (mkrds) [inline src/components/icon/icon.stories.js:24]: I think this knob should stay, as we decided to keep the Tooltip in Icon right?

File: src/components/icon/icon.stories.js [hunk 18]
```diff
@@ -19,16 +19,8 @@ export default {
...
     type: select("type", OptionsHelper.icons, "add"),
-    tooltipPosition: tooltipMessage
```
Comment (mkrds) [inline src/components/icon/icon.stories.js:26]: Same for this one

File: src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.component.js [hunk 19]
```diff
@@ -1,65 +1,54 @@
...
   onMouseDown: PropTypes.func.isRequired,
-  /** Callback to handle any mouse over events on a button */
-  onMouseOver: PropTypes.func.isRequired,
-  /** Callback to handle any mouse leave events on a button */
-  onMouseLeave: PropTypes.func.isRequired,
-  /** Controls whether the button's tooltip should be shown */
-  tooltipVisible: PropTypes.bool,
+  /** Controls whether the button can be tabbed to */
   tabbable: PropTypes.bool,
+  tooltipMessage: PropTypes.node,
```
Comment (mkrds) [inline src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.component.js:50]: You could add the description here

File: src/components/icon/icon.stories.js (hunk 17 above)
Comment (edleeks87) [reply src/components/icon/icon.stories.js:24]: yep good spot

File: src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.spec.js (hunk 7 above)
Comment (edleeks87) [reply src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.spec.js:103]: Ah actually I moved these tests into the Toolbar as it was where the buttons are wrapped in the `Tooltip`

File: src/components/tooltip/index.d.ts (hunk 10 above)
Comment (edleeks87) [reply src/components/tooltip/index.d.ts:1]: I thin it needs to point at the declaration file so `export { default } from './tooltip';` would be right

File: src/components/tooltip/index.d.ts (hunk 11 above)
Comment (edleeks87) [reply src/components/tooltip/index.d.ts:2]: yeah good point

File: src/components/tile-select/tile-select.stories.mdx [hunk 20]
```diff
@@ -91,7 +91,7 @@ These props are then internally distributed on each of the `TileSelects` making
             disabled
             title="Title" 
             subtitle="Subtitle"
-            titleAdornment={<Icon type='info' tooltipMessage="Short and non descriptive message" />}
+            titleAdornment={<Icon type='info' tooltipMessage="Short and non descriptive message" tooltipVisible={false} />}
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/tile-select/tile-select.stories.mdx:94]: It shouldn't display when the tile is disabled

File: src/components/tooltip/tooltip.component.js [hunk 21]
```diff
@@ -1,70 +1,97 @@
...
     return (
-      <StyledTooltipWrapper
-        role="tooltip"
-        onMouseEnter={onMouseEnter}
-        onMouseLeave={onMouseLeave}
-        {...commonProps}
-        {...tagComponent("tooltip", this.props)}
+      <Tippy
+        placement={position}
+        delay={100}
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/tooltip/tooltip.component.js:60]: Yeah good point

File: src/components/text-editor/__internal__/toolbar/toolbar.component.js [hunk 22]
```diff
@@ -56,89 +55,132 @@ const Toolbar = ({
...
+    [controlRefs, focusIndex, handleBlockType, handleInlineStyleChange]
   );
 
   useEffect(() => {
     if (focusIndex === null) {
       setTabbable(true);
-    } else if (canFocus && focusIndex !== null) {
-      controlRefs[focusIndex].current._target.focus();
     }
   }, [canFocus, controlRefs, focusIndex, tabbable]);
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/text-editor/__internal__/toolbar/toolbar.component.js:82]: They aren't, good spot

File: src/components/tile-select/tile-select.stories.mdx (hunk 20 above)
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/tile-select/tile-select.stories.mdx:94]: If you pass the `disabled` prop to `Icon` it adds additional styles that I wasn't sure was wanted so for now I've just fixed it for the purposes of this demo. Plus I didn't want to add more files to this PR 😄

File: src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.component.js [hunk 23]
```diff
@@ -1,65 +1,50 @@
...
-      </>
+      <StyledToolbarButton
+        data-component="text-editor-toolbar-button"
+        ref={ref}
+        onKeyDown={onKeyDown}
+        onMouseDown={onMouseDown}
+        isActive={activated}
+        aria-label={ariaLabel}
+        {...(!tabbable && { tabIndex: -1 })}
+        {...rest}
```
Comment (sjohnson-sage) [inline src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.component.js:27]: I don't think we just want to be passing down all of the props to the styled component here, only ones we specifically need

File: src/components/icon/icon.component.js [hunk 24]
```diff
@@ -100,6 +127,16 @@ Icon.propTypes = {
...
   /** Aria label for accessibility purposes */
   ariaLabel: PropTypes.string,
+  /** The message string to be displayed in the tooltip */
+  tooltipMessage: PropTypes.string,
+  /** The position to display the tooltip */
+  tooltipPosition: PropTypes.oneOf(["top", "bottom", "left", "right"]),
+  /** Control whether the tooltip is visible */
+  tooltipVisible: PropTypes.bool,
+  /** @ignore @private */
+  checked: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/icon/icon.component.js:139]: Have messaged privately, it's being passed somehow down from the `SimpleColor` component, I think it's safe to revert to spreading the `rest` here and tackle it in it's own ticket

File: src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.component.js (hunk 23 above)
Comment (edleeks87) [reply src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.component.js:27]: It's an internal component that's only composed within the TextEditor, but I'll update it for the sake of maintainability

File: src/__experimental__/components/label/label.spec.js [hunk 25]
```diff
@@ -17,6 +17,8 @@ import {
   InputGroupContext,
 } from "../../../__internal__/input-behaviour";
 
+jest.mock("@tippyjs/react/headless");
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/__experimental__/components/label/label.spec.js:20]: It's rendering a tippy state update warning when the tests run, I've had to add it to the `TileSelect` as well for the same reason

File: src/components/validations/validation-icon.component.js [hunk 26]
```diff
@@ -48,22 +50,31 @@ const ValidationIcon = ({
...
+      onBlur={() => setTriggeredByIcon(false)}
     >
-      <Icon
-        key={`${validationType}-icon`}
-        tooltipType={validationType}
-        tooltipMessage={validationMessage}
-        tooltipVisible={
-          hasFocus || hasMouseOver || groupHasFocus || groupHasMouseOver
-        }
+      <Tooltip
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/validations/validation-icon.component.js:58]: Makes sense, I wanted to avoid adding more props to `Icon` but I think you're right here so will remove

File: src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.d.ts (hunk 6 above)
Comment (edleeks87) [reply src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.d.ts:11]: The tooltip functionality is now handled in the `Toolbar`, I've added the event handler callbacks in though

**Review by sjohnson-sage — DISMISSED** [review]

File: src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.d.ts [hunk 27]
```diff
@@ -1,14 +1,15 @@
...
+  ariaLabel?: string;
   children: React.ReactNode;
-  activated: boolean;
+  activated?: boolean;
   onKeyDown: (event: React.KeyboardEvent<HTMLElement>) => void;
   onMouseDown: (event: React.MouseEvent<HTMLElement>) => void;
-  onMouseOver: (event: React.MouseEvent<HTMLElement>) => void;
-  onMouseLeave: (event: React.MouseEvent<HTMLElement>) => void;
-  tooltipVisible?: boolean;
+  onMouseOver?: (event: React.MouseEvent<HTMLElement>) => void;
```
Comment (mkrds) [inline src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.d.ts:9]: `tabbable` is missing

Comment (edleeks87) [reply src/components/text-editor/__internal__/toolbar/toolbar-button/toolbar-button.d.ts:9]: 🤦

**Review by mkrds — APPROVED** [review]

**Review by sjohnson-sage — DISMISSED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by KatarzynaQA — APPROVED** [review]
